### DISCUSSION TITLE
Does Durable recognise different JSON schema from same endpoint?

### DISCUSSION THREAD

**Discussion by yongsrepos** [postmanlabs/postman-flows#255] [Q&A] [answered]
Hi,

I have an endpoint that returns responses with different JSON schema, when the requests bodies are different(i.e., requestA -> responseA, requestB -> responseB).

Now I'm creating a Flows, that involves calling this endpoint twice, with requestA and requestB respectively. 
But when creating Durables on responseB, it can't recognise the extra JSON attributes in responseB. Instead, it only recognises the schema from responseA.

I've tried modify the Examples under both requests. But it didn't work.

Any help will be highly appreciated. Thanks!
BR,
Yong

**Comment by saswatds** [comment, marked as answer]
Hey @yongsrepos, currently the type-system performs an **intersection** on the keys to find out the safest object it can provide for consumption without null pointer errors. 
I have a few questions to improve the type system. Are schema of request A and request B completely different or,  is request A a subset of B?

**Reply by yongsrepos** [reply]
Thanks for the reply, @saswatds !
Regarding the requests (i.e., requestA and requestB), they have intersection fields, while each of them also has additional fields respectively. 
For example, while **requestA** is like: `{"a": "common", "b": "onlyInA"}`, then **requestB** is like: `{"a": "common", "c": "onlyInB"}`.

Regarding the responses, they are in the same situation as the requests.

**Reply by yongsrepos** [reply]
Hi @saswatds , now I've got a workaround to address the Flows creation problem. I copied the additional field from **responseB** to the Example of **responseA**. (Thanks to your hint on the **intersection** part.)

But there is problem running the Flows. It complains `Value "/body/diffFields" is a "symbol" primitive. Must be "complex" to have field "diffFields"`. Any hint? Thanks in advance!
/Yong

**Reply by saswatds** [reply]
Hey @yongsrepos , could you give me a hint as to what the expression looks like? The only symbol within the system right now is `null` so maybe the data is null and something is failing. If I get a hint of the expression, it will be easier for me to find the bug.

**Reply by yongsrepos** [reply]
Hi @saswatds , sorry I don't know what you are referring to with `expression`. Could you elaborate? Thanks!
Besides, above error is from `Execution issues` from the main window.

**Reply by yongsrepos** [reply]
If you are referring to the responses JSON schema, then for **responseA**, it's like: `{"a": "common", "aObj": {"fieldA1":"onlyInA"}}`, then **responseB** is like: `{"a": "common", "bObj": {"fieldB1":"onlyInB"}}`.

**Reply by saswatds** [reply]
Got it. I have been able to find the point of failure. I will try and find out a solution to this and push out a quick bug fix. The runtime error is happening as for some reason one of the elements in the path has become null. 

Thanks a lot of reporting this issue.

**Reply by yongsrepos** [reply]
Thanks a lot! Look forward to the fix!

**Reply by yongsrepos** [reply]
Hi @saswatds , is there a GitHub issue around this topic that I can follow the status? Really hope it will be fixed. It will be so helpful for us to verify an end-2-end flow automatically. Thanks a lot!

**Reply by yongsrepos** [reply]
Good day, @saswatds . FYI, I've both contacted Postman support from my company's account and created a Postman app issue here: https://github.com/postmanlabs/postman-app-support/issues/10747

**Reply by saswatds** [reply]
@yongsrepos It's on my immediate to-do list, will keep you posted through the support channel.

**Reply by yongsrepos** [reply]
Perfect! Thanks, @saswatds .
